code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java

```diff
@@ -30,17 +31,34 @@
 import io.cattle.platform.object.util.DataAccessor;
 
 import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
 
 import org.jooq.JoinType;
 
 public class MetaDataInfoDaoImpl extends AbstractJooqDao implements MetaDataInfoDao {
+
+    @Inject
+    DnsInfoDao dnsInfoDao;
+
     @Override
-    public List<ContainerMetaData> getContainersData(long accountId) {
+    public List<ContainerMetaData> getContainersData(final long accountId) {
         MultiRecordMapper<ContainerMetaData> mapper = new MultiRecordMapper<ContainerMetaData>() {
             @Override
             protected ContainerMetaData map(List<Object> input) {
                 ContainerMetaData data = new ContainerMetaData();
-                data.setIp((IpAddress) input.get(3));
+
+                Instance instance = (Instance) input.get(1);
+
+                final Map<Long, IpAddress> instanceIdToHostIpMap = dnsInfoDao
```

Comment:
@prachidamle , instanceIdToHostIpMap map should be fetched just once. If we place it here, it would be fetched every time we transform map to object, basically for every record. Move it up to the very beginning of the method.